langchain/vectorstores/elastic_vector_search.py

```diff
@@ -20,10 +20,15 @@ def _default_text_mapping(dim: int) -> Dict:
     }
 
 
-def _default_script_query(query_vector: List[float]) -> Dict:
+def _default_script_query(query_vector: List[float], filter: dict) -> Dict:
+    if filter:
+        (key, value), = filter.items()
```

Comment:
could there be a valid filter with more than one key/val pair? if not could we explicitly check filter length

Reply:
![github_1](https://user-images.githubusercontent.com/26499743/232938588-0722497a-6d7a-4771-ba27-eea4fbf17f89.png)

Pdfloader (PyMuPDF) extracts metadata including source, file path, page number, total pages, format, title, author, keywords, creator, producer, creation date, mod date, and trapped.

I find filtering by file path is very useful. I am a newbie, Perhaps you could demonstrate how to compile that for me.